.github/workflows/azureml-cpu-nightly.yml

```diff
@@ -67,7 +67,7 @@ jobs:
     strategy:
       max-parallel: 50 # Usage limits: https://docs.github.com/en/actions/learn-github-actions/usage-limits-billing-and-administration
       matrix:
-        python-version: ["3.8", "3.9", "3.10", "3.11"]
+        python-version: ["3.9", "3.10", "3.11", "3.12"]
```

Comment:
@daviddavo @SimonYansenZhao 

A good way to decide whether or not to drop 3.8 is to look at the percentage of people using it. We can take a percentage and say for example, if only 5% of people use 3.8 then we drop it, otherwise, we support it. Do you know where we can find that percentage?

Reply:
Lets also consideer stats for this package: 

According to [pypistats](https://pypistats.org/packages/recommenders), its about 10% and more used than 3.11... But every day more and more deps are dropping it. You can always use an old version of this package.

![image](https://github.com/user-attachments/assets/11d09911-3e62-419e-b79c-de39a9ed15ab)

Nevertheless, I can try and relax the version of numpy and see if the tests pass.

Reply:
I just found it.  This is from PyPI's download statistics.
<img width="656" alt="Screenshot 2024-08-01 at 14 49 27" src="https://github.com/user-attachments/assets/afddb505-b3e0-4ef9-b2d4-685ba26268cd">

Reply:
Ill solve the breaking changes in tensorflow and then revert the "removing 3.8" commit

Reply:
It seems now it's the second most used python. I think we should support it for now. 
One way is to install some dependencies for more than 3.8 and another for less than 3.8. Like pandera here: https://github.com/recommenders-team/recommenders/blob/4f86e4785337d455aa3cb7e8920c3fab9a2a0140/setup.py#L41